### PR TITLE
Fix covscan issues

### PR DESCRIPTION
**Pull request by jan-cerny**
Addressing some covscan issues in RHEL9. See the commit messages for details.

### CODE REVIEW

File: tests/API/crypt/test_crapi_digest.c [hunk 1]
```diff
@@ -40,6 +40,10 @@ static int mem2hex (uint8_t *mem, size_t mlen, char *str, size_t slen)
 {
         const char ch[] = "0123456789abcdef";
         register size_t i;
+
+	if (str == NULL) {
```
Comment (evgenyz) [inline tests/API/crypt/test_crapi_digest.c:44]: Can we please have consistent indentation here?

Comment (jan-cerny) [reply tests/API/crypt/test_crapi_digest.c:44]: I can't find which line is indented wrongly. I can see that all 3 indented lines are indented by tabs.

Comment (evgenyz) [reply tests/API/crypt/test_crapi_digest.c:44]: Yeah, but the code around them is indented with spaces. You can see it even in the GH diff. It is really painful to skim through such a code. As far as I remember we agreed on keeping the indentation style of the original code if changes are small.

**Review by evgenyz — APPROVED** [review]
Nice! Thank you!

**Pull request merged**